lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -200,8 +203,13 @@ export class AuthorizationCodeClient extends BaseClient {
         // Deserialize hash fragment response parameters.
         const hashUrlString = new UrlString(hashFragment);
         // Deserialize hash fragment response parameters.
-        const serverParams: ServerAuthorizationCodeResponse =
-            UrlString.getDeserializedHash(hashUrlString.getHash());
+        let serverParams : ServerAuthorizationCodeResponse;
+        if(this.config.authOptions.authority.options.OIDCOptions?.serverResponseType === ServerResponseType.QUERY) {
+            serverParams = UrlString.getDeserializedHash(hashFragment);
```

Comment:
`getDeserializedQueryString`?

Reply:
I'd vote for `parseQueryString` instead. If the method doesn't already exist, which I'm fairly certain it does.